core/src/modules/caches/appsCache.ts

```diff
@@ -0,0 +1,52 @@
+import { ModelCache } from "../modelCache";
+import { App } from "../../models/App";
+import { Option } from "../../models/Option";
+import { Includeable, WhereAttributeHash } from "sequelize";
+
+function getInclude() {
+  const include: Includeable[] = [{ model: Option, required: false }];
+  return include;
+}
+
+async function findAllWithCache(
+  this: ModelCache<App>,
+  modelId?: string,
+  state?: App["state"]
+) {
+  const now = new Date().getTime();
+  if (this.expires > now && this.instances.length > 0) {
+    return this.instances.filter((a) => !state || a.state === state);
+  } else {
+    this.instances = await App.unscoped().findAll({ include: getInclude() });
+    this.expires = now + this.TTL;
+    return this.instances.filter((a) => !state || a.state === state);
+  }
+}
+
+async function findOneWithCache(
+  this: ModelCache<App>,
+  value: string,
+  modelId?: string,
+  state?: App["state"],
+  lookupKey: keyof App = "id"
+) {
+  const instances = await this.findAllWithCache();
+  let instance = instances.find((i) => i[lookupKey] === value);
+
+  if (!instance) {
+    const where: WhereAttributeHash = { [lookupKey]: value };
+    if (state) where.state = state;
```

Comment:
I think this should work fine (if typescript doesn't complain) since it'll either be what we want, or undefined and thus ignored.
```suggestion
    where.state = state;
```

Reply:
the problem is then that you pass in either an undefined (literal) or null - both of which sequelize will try to stringily and use, because the key on the where object exists.